Products/CMFPlone/_scripts/verifydb.py

```diff
@@ -0,0 +1,88 @@
+from Zope2.Startup.run import make_wsgi_app
+from ZODB.interfaces import IStorageCurrentRecordIteration
+from ZODB.serialize import PersistentUnpickler
+import argparse
+import io
+import logging
+import pdb
+import pickletools
+import sys
+import traceback
+import Zope2
+
+logger = logging.getLogger('zodbverify')
+
+
+def zopectl_entry(self, arg):
+    parser = argparse.ArgumentParser(
+        prog=sys.argv[0] + ' verifydb',
+        description='Verifies that all records in the database can be loaded.',
+    )
+    parser.add_argument(
+        '-D', '--debug', action='store_true', dest='debug',
+        help='pause to debug broken pickles')
+    options = parser.parse_args(arg.split(' ') if arg else [])
+
+    logging.basicConfig(level=logging.INFO)
+    make_wsgi_app({}, self.options.configfile)
+    app = Zope2.app()
+    verify_zodb(app, debug=options.debug)
+
+
+def verify_zodb(obj, debug=False):
+    storage = obj._p_jar._db._storage
+    if not IStorageCurrentRecordIteration.providedBy(storage):
+        raise TypeError(
+            'ZODB storage {} does not implement record_iternext'.format(
+                storage))
+
+    logger.info('Scanning ZODB...')
+
+    next = None
```

Comment:
`next` is a builtin, please use a different variable name.

Reply:
Ah yes, thanks. I added https://github.com/zopefoundation/ZODB/issues/233 to fix the example I based this on.